ostd/src/arch/x86/mod.rs

```diff
@@ -177,7 +178,48 @@ fn has_avx512() -> bool {
 }
 
 pub(crate) fn enable_cpu_features() {
-    use x86_64::registers::{control::Cr4Flags, model_specific::EferFlags, xcontrol::XCr0Flags};
+    use x86::cpuid;
+    use x86_64::registers::{
+        control::{Cr0Flags, Cr4Flags},
+        model_specific::EferFlags,
+        xcontrol::XCr0Flags,
+    };
+
+    CPU_FEATURES.call_once(|| {
+        let cpuid = cpuid::CpuId::new();
+        cpuid.get_feature_info().unwrap()
+    });
+
+    XSTATE_MAX_FEATURES.call_once(|| {
```

Comment:
Do not expose `XSTATE_MAX_FEATURES` and `XSAVE_SIZE` from the `cpu` module. Instead, add an `enable_essential_features` features to the `cpu` module and call the method here.

```rust
pub(crate) fn enable_cpu_features() {
    CPU_FEATURES.call_once(|| { ... })

    cpu::enable_essential_features();

    // ..
}
```

This way we hide details from the top-level CPU feature enabling function.